src/main/java/game/Board.java

```diff
@@ -254,7 +253,7 @@ private long attacksTo(int sq, boolean attacker, long occupied) {
             Bitboard.pawnAttacks(!attacker, sq) & this.pawns);
     }
 
-    public void legalMoves(ArrayList<Move> moves) {
+    public void legalMoves(MoveList moves) {
```

Comment:
Perhaps further performance improvement (of this code) is possible through usage of the Java 8 Streams API?